backend/src/schema/resolvers/searches.js

```diff
@@ -3,90 +3,200 @@ import { queryString } from './searches/queryString'
 
 // see http://lucene.apache.org/core/8_3_1/queryparser/org/apache/lucene/queryparser/classic/package-summary.html#package.description
 
+const cypherTemplate = (setup) => `
+  CALL db.index.fulltext.queryNodes('${setup.fulltextIndex}', $query)
+  YIELD node AS resource, score
+  ${setup.match}
+  ${setup.whereClause}
+  ${setup.withClause}
+  RETURN 
+  ${setup.returnClause}
+  AS result
+  SKIP $skip
+  ${setup.limit}
+`
+
+const simpleWhereClause =
+  'WHERE score >= 0.0 AND NOT (resource.deleted = true OR resource.disabled = true)'
+
+const postWhereClause = `WHERE score >= 0.0
+  AND NOT (
+    author.deleted = true OR author.disabled = true
+    OR resource.deleted = true OR resource.disabled = true
+    OR (:User {id: $userId})-[:MUTED]->(author)
+  )`
+
+const searchPostsSetup = {
+  fulltextIndex: 'post_fulltext_search',
+  match: 'MATCH (resource)<-[:WROTE]-(author:User)',
```

Comment:
I would prefer to have the type on the `resource ` for clearness and better readability:
`match: 'MATCH (resource:Post)<-[:WROTE]-(author:User)',`